bindings/ruby/ext/regorusrb/src/lib.rs

```diff
@@ -0,0 +1,213 @@
+use magnus::{exception::runtime_error, method, module, prelude::*, Error, Ruby};
+use regorus::Engine as RegorusEngine;
+use std::cell::RefCell;
+use std::cmp::Ordering;
+
+// `Value` exists under magnus, regorus, and serde_json, so be explicit
+
+#[derive(Default)]
+#[magnus::wrap(class = "Regorus::Engine")]
+pub struct Engine {
+    engine: RefCell<RegorusEngine>,
```

Comment:
Why do we need a RefCell?

Reply:
I think because we need an immutable struct to expose to Ruby, but we want a mutable reference to RegorusEngine to call add_data or eval_rule? I'm a Rust noob haha, doing my best to follow the magnus [docs](https://github.com/matsadler/magnus?tab=readme-ov-file#wrapping-rust-types-in-ruby-objects)

The other option of passing a `&mut self` as the first argument to those methods doesn't appear to work with the magnus macros.

Reply:
Makes sense. Thanks for the clarification,